File: LeetCode/Simulation/JS/Leetcode1331-rank-transform-of-an-array.js
var arrayRankTransform = function(arr) {
    let n = arr.length;
    let tmp = [...arr];
    tmp.sort((a,b) => a - b);
    let mp = new Map();
    for(let [i,cnt] = [0,1];i < n;i++) {
        if(!(mp.get(tmp[i]) | 0)) {
            mp.set(tmp[i],cnt);
            cnt++;
        }
    }
    let ans = new Array(n).fill(0);
    for(let i = 0;i < n;i++) {
        ans[i] = mp.get(arr[i]);
    }
    return ans;
};

File: LeetCode/Simulation/Rust/Leetcode1331-rank-transform-of-an-array.rs
use std::collections::HashMap;
impl Solution {
  pub fn array_rank_transform(arr: Vec<i32>) -> Vec<i32> {
    let (n, mut tmp) = (arr.len(), arr.clone());
    tmp.sort();
    let mut mp = HashMap::new();
    let mut cnt = 1;
    for i in 0..n {
      if mp.contains_key(&tmp[i]) {
        continue;
      }
      mp.insert(tmp[i], cnt);
      cnt += 1;
    }
    let mut ans = vec![0; n];
    for i in 0..n {
      ans[i] = mp[&arr[i]]
    }
    ans
  }
}
